src/sentry_database.c

```diff
@@ -173,15 +173,35 @@ sentry__process_old_runs(const sentry_options_t *options)
         const sentry_path_t *file;
         while ((file = sentry__pathiter_next(run_iter)) != NULL) {
             if (sentry__path_filename_matches(file, "session.json")) {
+                if (!session_envelope) {
+                    session_envelope = sentry__envelope_new();
+                }
                 sentry_session_t *session = sentry__session_from_path(file);
                 if (session) {
+                    // this is just a heuristic: whenever the session was not
+                    // closed properly, and we do have a crash that happened
+                    // *after* the session was started, we will assume that the
+                    // crash corresponds to the session and flag it as crashed.
+                    // this should only happen when using crashpad, and there
+                    // should normally be only a single unclosed session at a
+                    // time.
```

Comment:
Are we already in a new session when this is invoked?

Reply:
No, we auto-start the session as the last thing, after processing the old runs. Either way this should be completely independent.